lib/ansible/modules/apt.py

```diff
@@ -687,23 +711,21 @@ def install(m, pkgspec, cache, upgrade=False, default_release=None,
         name, version_cmp, version = package_split(package)
         package_names.append(name)
         installed, installed_version, version_installable, has_files = package_status(m, name, version_cmp, version, default_release, cache, state='install')
-        if (not installed and not only_upgrade) or (installed and not installed_version) or (upgrade and version_installable):
-            if version_installable or version:
-                pkg_list.append("'%s=%s'" % (name, version_installable or version))
+        if (not installed and not only_upgrade) or (installed and not installed_version and version_installable) or (upgrade and version_installable):
```

Comment:
I'm not sure I read this logic properly, or well, fully see all cases when version will be specified for the package installation (either because of my skills or overall code readability), so having some comment about intended behavior would be useful here IMO.